### ISSUE TITLE
Streaming just dropped

### ISSUE THREAD

**Issue by davidbrowne17** [resemble-ai/chatterbox#73]
https://github.com/davidbrowne17/chatterbox-streaming
about 0.5s to first audio chunk on a 4090. Enjoy <3

**Comment by websines** [comment]
Hi @davidbrowne17  how were you able to get the RTF down by so much?

Edit: Nvm too slow for me. On a 3090.
Latency to first chunk: 4.771s
Chunk 1
Chunk 2
Chunk 3
Chunk 4
Chunk 5
Chunk 6
Total generation time: 16.284s
Total audio duration: 5.720s
RTF (Real-Time Factor): 2.847
Total chunks yielded: 6

Also noticed 12gb of VRAM being used, did I break it lol

**Comment by davidbrowne17** [comment]
> Hi [@davidbrowne17](https://github.com/davidbrowne17) how were you able to get the RTF down by so much?
> 
> Edit: Nvm too slow for me. On a 3090. Latency to first chunk: 4.771s Chunk 1 Chunk 2 Chunk 3 Chunk 4 Chunk 5 Chunk 6 Total generation time: 16.284s Total audio duration: 5.720s RTF (Real-Time Factor): 2.847 Total chunks yielded: 6
> 
> Also noticed 12gb of VRAM being used, did I break it lol

Are you using Windows? Its a lot slower than linux. Make sure you are building torch with cuda, the pip install doesnt do that for windows

**Comment by websines** [comment]
Oh yeah doing windows, but that much of a difference normal? I built torch with cuda 12.8

**Comment by davidbrowne17** [comment]
> Oh yeah doing windows, but that much of a difference normal? I built torch with cuda 12.8

Yeah its a big difference, linux is a lot faster at cuda operations, if you run more than a single generation it should speed up a bit as the model will be warmed up its about 1.0 RTF after warmup on windows

**Comment by websines** [comment]
I used your CSM one as well, do you recommend that one of this. Which uses less vram?

**Comment by davidbrowne17** [comment]
> I used your CSM one as well, do you recommend that one of this. Which uses less vram?

It kinda depends on your use case, this model uses a bit less vram but not by too much, CSM is better but only if you have enough data to fine-tune it (at least an hour), if you want one shot cloning without having to train use this one. CSM is currently faster but this one hasn't been optimized too much yet

**Comment by patientx** [comment]
So the "pauses" between chunks are not avoidable , am I right ?

**Comment by davidbrowne17** [comment]
> So the "pauses" between chunks are not avoidable , am I right ?

Increase the chunk_size in the generate_stream method to whatever value makes it smooth, itll vary depending on your gpu

**Comment by TediPapajorgji** [comment]
@davidbrowne17 is back at it again!
